### PR TITLE
Add caching on layer tree

### PR DESCRIPTION
**Pull request by SebCorbin**

### CODE REVIEW

File: terra_layer/views.py [hunk 1]
```diff
@@ -54,10 +55,11 @@ def get(self, request, slug=None, format=None):
             raise Http404('View does not exist')
 
         view = settings.TERRA_LAYER_VIEWS[slug]
-        layers = self.layers(view['pk'])
 
-        return Response(
-            {
+        response_dict = cache.get('terra-layer-{}'.format(view['pk']))
```
Comment (nahuelange) [inline terra_layer/views.py]: Could you create a method to get the layer's cache key to unify its formating?

Comment (nahuelange) [reply terra_layer/views.py]: Then, maybe cache.get_or_set could be used here, don't you think ?

File: terra_layer/models.py [hunk 2]
```diff
@@ -68,6 +70,17 @@ class Meta:
             ('can_manage_layers', 'Can manage layers'),
         )
 
+    def save(self, **kwargs):
+        super().save(**kwargs)
+
+        # Find the upmost group for this layer
+        group = self.group
+        if group:
+            while group.parent:
```
Comment (nahuelange) [inline terra_layer/models.py]: You don't need to go up in the group tree, you can you group.view

Comment (SebCorbin) [reply terra_layer/models.py]: What if tree is nested? Like on https://terralego.netlify.com/?path=/story/modules-visualizer-layerstree--component

Comment (nahuelange) [reply terra_layer/models.py]: Every group has a view attribute, that in which view it appears, event children has this view attribute.

File: terra_layer/views.py (hunk 1 above)
Comment (SebCorbin) [reply terra_layer/views.py]: Added through a new method

Comment (nahuelange) [reply terra_layer/views.py]: :+1:

File: terra_layer/models.py (hunk 2 above)
Comment (nahuelange) [reply terra_layer/models.py]: :+1:

**Pull request merged**